remediation/workflow/pin/pinactions.go

```diff
@@ -69,15 +69,29 @@ func PinAction(action, inputYaml string, exemptedActions []string, pinToImmutabl
 	tc := oauth2.NewClient(ctx, ts)
 
 	client := github.NewClient(tc)
+	var commitSHA string
+	var err error
 
-	commitSHA, _, err := client.Repositories.GetCommitSHA1(ctx, owner, repo, tagOrBranch, "")
-	if err != nil {
-		return inputYaml, updated
-	}
+	if actionCommitMap != nil && actionCommitMap[action] != "" {
```

Comment:
Are GitHub Action references case sensitive? If not, this will not work in case the action is referenced in different case (e.g., actionorg/actionname@v2 and ActionOrg/Actionname@v2 will be treated differently in the current appraoch). If GitHub Action references are case insensitive, we should create a struct and perform case insensitive comparison on action name and/or version tag.